src/Workspaces/Core/Desktop/Workspace/Host/Documentation/DocumentationProviderServiceFactory.cs

```diff
@@ -29,15 +30,46 @@ public DocumentationProvider GetDocumentationProvider(string assemblyPath)
                     throw new ArgumentNullException(nameof(assemblyPath));
                 }
 
-                assemblyPath = Path.ChangeExtension(assemblyPath, "xml");
+                string xmlDocumentationFilePath;
+                if (!TryFindDocumentationFile(assemblyPath, out xmlDocumentationFilePath))
+                {
+                    return _assemblyPathToDocumentationProviderMap.GetOrAdd(nameof(DocumentationProvider.Default), DocumentationProvider.Default);
```

Comment:
Hmm.. The key is a string, typically path to the xml file, only that, in this case the file doesnt exist. Also, "for any" xml file that doesn't exist we want to just use same cached NullDocumentationProvider, which is what I've done here.. I didn't see the benefit of having multiple entries in table with various keys having the same value (null provider in this case).